tests/deprecated_api/test_remove_2-0.py

```diff
@@ -11,12 +11,39 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Test deprecated functionality which will be removed in v2.0."""
+"""Test deprecated functionality which will be removed in v2.0.0."""
+from unittest import mock
+
 import pytest
 
 from pytorch_lightning import Trainer
 from tests.callbacks.test_callbacks import OldStatefulCallback
 from tests.helpers import BoringModel
+from tests.helpers.runif import RunIf
+
+
+def test_v2_0_0_deprecated_num_processes(tmpdir):
+    with pytest.deprecated_call(match=r"is deprecated in v1.6 and will be removed in v2.0."):
+        _ = Trainer(default_root_dir=tmpdir, num_processes=2)
+
+
+@mock.patch("torch.cuda.is_available", return_value=True)
+@mock.patch("torch.cuda.device_count", return_value=2)
+def test_v2_0_0_deprecated_gpus(mock_is_available, mock_device_count, tmpdir):
+    with pytest.deprecated_call(match=r"is deprecated in v1.6 and will be removed in v2.0."):
+        _ = Trainer(default_root_dir=tmpdir, gpus=2)
+
+
+@RunIf(tpu=True)
```

Comment:
Could you mock it please for tpus and ipu?

Reply:
do we have any examples of this?

Reply:
I tried adding `@mock.patch("pytorch_lightning.utilities.imports._TPU_AVAILABLE", return_value=True)` but am still getting `MisconfigurationException: No TPU devices were found`

Reply:
This PR https://github.com/PyTorchLightning/pytorch-lightning/pull/12326 will make the mocking easier.

Reply:
Ok, thanks. It seems for IPUs it can't be mocked either

Reply:
If you want to mock that constant, you need to mock it where it is used, not where it is defined. 
Why? Because it's basically impossible to mock it BEFORE any other module imports it, and in the test you don't have influence on that. So, in short, I recommend `monkeypatch.setattr(themodule_where_it_is_used, "_TPU_AVAILABLE", True)`. 

Hope this helps.

Reply:
thanks @awaelchli! I tried that but keep running into other issues, like `NameError: name 'xm' is not defined`. I think it probably easier to just leave this unmocked. If someone else wants to try and mock this for ipus and tpus, please feel free to add a commit!

Reply:
@daniellepintz, when single tpu core is used, it will use SingleTPUStrategy, which calls `xm.xla_device` at initialization and make things harder to mock. It's easier to just use 8 cores instead.

Not sure how to add a commit to your PR directly. You can checkout https://github.com/DuYicong515/pytorch-lightning/commit/1693091764910d3b849e1f004581d11027906bad as a reference.

Reply:
Thanks @DuYicong515, yes I think that's the way :) 
@daniellepintz note that the test would otherwise not run, as we don't collect this TPU test (even if it's marked as `RunIf(tpu=True)`. #11098

Reply:
thank you so much @DuYicong515! That worked!!